<?xml version="1.0"?>
<!--apriltag检测-->
<launch>
    <arg name="use_depth_cam"       default="true"/>
    <arg name="usb_cam_name"        default="usb_cam"/>
    <arg name="depth_camera_name"   default="depth_cam"/>
    <arg name="launch_prefix"       default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="node_namespace"      default="apriltag_ros_continuous_node" />
    <arg name="image_topic"         default="image_raw" />

    <!--USB摄像头-->
    <include unless="$(arg use_depth_cam)" file="$(find jetauto_peripherals)/launch/usb_cam.launch">
        <arg name="usb_cam_name" value="$(arg usb_cam_name)"/>
    </include>

    <!--深度摄像头-->
    <include if="$(arg use_depth_cam)" file="$(find jetauto_peripherals)/launch/depth_cam.launch">
        <arg name="depth_camera_name" value="$(arg depth_camera_name)"/>
    </include>

    <!--底盘驱动(chassis driver)-->
    <include file="$(find jetauto_controller)/launch/jetauto_controller.launch"/>
    <node name="init_pose" pkg="jetauto_slam" type="init_pose.py" output="screen">
        <param name="horizontal" value="true"/>
    </node>

    <!-- Set parameters -->
    <rosparam command="load" file="$(find jetauto_example)/config/settings.yaml" ns="$(arg node_namespace)" />
    <rosparam command="load" file="$(find jetauto_example)/config/tags.yaml"     ns="$(arg node_namespace)" />

    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap if="$(arg use_depth_cam)" from="image_rect"  to="/$(arg depth_camera_name)/rgb/$(arg image_topic)" />
        <remap if="$(arg use_depth_cam)" from="camera_info" to="/$(arg depth_camera_name)/camera_info" />
        <remap unless="$(arg use_depth_cam)" from="image_rect"  to="/$(arg usb_cam_name)/$(arg image_topic)" />
        <remap unless="$(arg use_depth_cam)" from="camera_info" to="/$(arg usb_cam_name)/camera_info" />
        <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find jetauto_example)/rviz/apriltag.rviz" />
</launch>
